server/src/main/java/org/elasticsearch/Build.java

```diff
@@ -130,6 +130,40 @@ public static String minimumCompatString(IndexVersion minimumCompatible) {
         }
     }
 
+    public static boolean isWireCompatibleWithCurrent(String version) {
+        try {
+            TransportVersion transportVersion = TransportVersion.fromString(version);
+            return transportVersion.onOrAfter(TransportVersions.MINIMUM_COMPATIBLE);
+        } catch (NumberFormatException | IllegalStateException e) {
+            // not a transport version, continue
+        }
+
+        try {
+            Version esVersion = Version.fromString(version);
+            return esVersion.onOrAfter(Version.CURRENT.minimumCompatibilityVersion());
```

Comment:
We should be logging/notifying on calls that get to this point, we want to phase out Version from transport & index complete.y

Reply:
I think both of these will be called in the context of loading previous-version node metadata from disk, though I'll have to take a closer look to be 100% confident. I don't know if we ought to log in that situation; we might get a lot of logs on certain upgrades. Should this method have an argument like `allowLegacyVersions` so we can throw an exception in other contexts?

Reply:
Yeah, maybe have a separate method for instances where we know we need to read historic versions - that way we isolate `Version` to where it is actually still needed

Reply:
Sounds good!